File: codes/go/custom_error_and_error_handling/custom_error_and_error_handling.go
package main

import (
	"errors"
	"fmt"
)

func giveMeRespect(argument string) (string, error) {
    if (argument != "respect") {
        return "", errors.New("Error: You should give me \"respect\"!")
    }
    return "Thank you for giving me \"respect\"!", nil
}

func main() {
    fmt.Println("\n// Custom Error and Error Handling in Go")

    // Error Handling
    response, anErrorMessage := giveMeRespect("boo!")
    if (anErrorMessage == nil) {
        fmt.Println(response)
    } else {
        fmt.Println(anErrorMessage)
    }
    fmt.Println("I'm sorry!")
}

File: codes/visual_basic_dotnet/custom_error_and_error_handling.vb
Imports System.Collections.Generic ' Dictionary, KeyValuePair

Module Program
    Function GiveMeRespect(ByVal Argument As String) As String
        ' Custom Error
        If (Argument <> "respect") Then Throw New Exception("Exception: You should give me ""respect""!")
        Return "Thank you for giving me ""respect""!"
    End Function

    Sub Main(Args As String())
        Console.WriteLine($"{Environment.NewLine}' Custom Error and Error Handling in Visual Basic (.NET)")

        Dim Response As String

        ' Error Handling
        Try
            Response = GiveMeRespect("boo!")
            Console.WriteLine(Response)
        Catch AnException As Exception
            Console.WriteLine(AnException.Message)
        End Try
        Console.WriteLine("I'm sorry!")

        ' Without Error Handling
        Response = GiveMeRespect("boo!")
        ' this will never executed
        Console.WriteLine(Response)
        Console.WriteLine("I'm sorry!")
    End Sub
End Module
